### PR TITLE
chore(security): flesh out security policy

### LINKED ISSUE #1551: Add security page to Vector docs
We need a document that outlines the various practices Vector takes to ensure the code is secure. We should add this to [SECURITY.md](https://github.com/timberio/vector/blob/master/SECURITY.md), but this is more of a formality for [Github's security policy feature](https://github.com/timberio/vector/security/policy).

See [this user note](https://timber.productboard.com/insights/my-inbox/notes/7404163).

### PR DESCRIPTION
**Pull request by lukesteensen**
Closes #1551 

Do we want to link to this from the docs, or have a page there with similar content? /cc @binarylogic

### CODE REVIEW

**Comment by binarylogic** [comment]
Looks good! Thanks for taking care of this. I think it makes where you have it (`SECURITY.md`), especially given that it's made available [here](https://github.com/timberio/vector/security/policy). Most importantly, I want to make sure that it answers these questions:

• software design (to see if they consider security requirements in the architecture and design phase of the product and new features)
• code reviews (to see if they happen and security checks are part of it),
• how changes are merged (to see if changes need to be approved and are well controlled),
• how dependencies are managed (to see if they scan for vulnerable dependencies, how they review dependencies before adding them to the product and how vulnerable or non-maintained dependencies are handled),
• how the source code is checked for vulnerabilities (e.g. static code analysis, penetration tests), etc.
• how the build process is secured (to see if they prevent modifications during the process)

I believe it does. I'll review it more in-depth tomorrow.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,7 +69,13 @@ Please ensure your commits are small and focused; they should tell a story of
 your change. This helps reviewers to follow your changes, especially for more
 complex changes.
 
-#### Signing
+#### Cryptographically Signing
+
+Vector requires all commits to be cryptographically signed as part of our
```
Comment (lukesteensen) [inline CONTRIBUTING.md]: Do we really? This is not something I've been doing and seems like a potentially large hurdle for accepting contributions.

File: SECURITY.md [hunk 2]
```diff
@@ -1,19 +1,229 @@
...
+Once passed, the pull request must be squashed and merged. This creates a clean
+linear history with a Vector team member's co-sign.
+
+#### Automated Checks
+
+When possible, we'll create automated checks to enforce security policies.
+
+##### Vulnerability Scans
+
+Vector implements an automated [`cargo audit` check][urls.cargo_audit]. This
```
Comment (lukesteensen) [inline SECURITY.md]: We actually use `cargo deny` for this. The database is the same.

File: SECURITY.md [hunk 3]
```diff
@@ -1,19 +1,229 @@
...
+Vector's code from the relevant [protected branch](#protected-branches),
+Docker image retrieval, and publishment of Vector's release artifacts.
+
+### Runtime Isolation
+
+All builds run in an isolated sandbox that is destroyed after each use.
+
+### Asset Encryption
+
+Vector's artifacts are stored on S3 with SSE encryption enabled.
```
Comment (lukesteensen) [inline SECURITY.md]: This seems unnecessary. The artifacts are not confidential, so I don't see how encrypting them on S3 adds anything.

File: SECURITY.md [hunk 4]
```diff
@@ -1,19 +1,229 @@
...
+
+#### Unsafe Code
+
+Vector does not allow the use of unsafe code except in circumstances where it
+is required, such as dealing with CFFI.
+
+#### User Privileges
+
+Vector is always designed to run under non-`sudo` privileges, and our
+documentation always defaults to non-`sudo` use.
```
Comment (lukesteensen) [inline SECURITY.md]: Probably better to use `root` rather than `sudo`. You can `sudo` to any user.

File: REVIEWING.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+defined in the [`.github/CODEOWNERS`](.github/CODE_OWNERS) file. Involving code
+owners ensures all context is applied to reviews since they have intimate
+knowledge of the code.
+
+## Dependencies
+
+Dependencies should be _carefully_ selected. Before adding a dependency we
+should ask the following questions:
+
+1. Is the dependency necessary? Can we avoid it?
```
Comment (lukesteensen) [inline REVIEWING.md]: This seems a little strong. We could always avoid the dependency by rewriting it ourselves, so it's more a question of whether or not the benefit is worth the cost.

**Pull request merged**